### PR TITLE
Fix RA/TD selection/decoration/interaction bounds

### LINKED ISSUE #15184: #15148 Regressed Auto-Calculated Selection Boxes
#15148
Here is RA War Factory for example:

resim

Selection boxes also different during and after make anim for some buildings.

### PR DESCRIPTION
**Pull request by reaperrr**
The dynamic/per-frame 'canvas' for SHP(TD) introduced by #15148 messed up the old auto-calculation assumptions.

We already use fixed bounds in most other places (D2k, TS, all moving actors in RA and TD, several structures in RA/TD), so finishing the job for RA/TD buildings is the easiest way to fix #15184, avoiding both reversion of #15148 as well as code-level work-arounds.

### CODE REVIEW

**Comment by pchote** [comment]
Do we need to do civilian buildings and tech structures too?

**Comment by lawando** [comment]
Tested the changes in ra, cnc and ts. In ra and cnc the bounds for civilian buildings/tech buildings are fine, in ts only the scrin ship needs adjustments. 

See https://imgur.com/a/VbWDHB8 for screenshots of selected civilian buildings per mod. Note that those buildings in the screenshots that are not selected (couldn't select them for some reason) have correct bounds too.

**Comment by reaperrr** [comment]
Added RA/TD tech buildings to be on the safe side.

@lawando It could be (but I haven't checked yet) that the damaged frame of the civ buildings alters the bounds a bit. That's pretty much the only possible issue to look out for, though.

TS Scrin ship should look like that before #15148 already.
TS buildings should get isometric selection-bounds/-boxes eventually, with the former derived from just the footprint and the latter calculated from footprint + a manually set height, just like in the original, so anything that passes as "acceptable" is good enough for TS buildings for now, because that will be adressed properly later on anyway.

**Comment by pchote** [comment]
I have added a section to https://github.com/OpenRA/OpenRAModSDK/wiki/Update-notes:-release-20180307-to-next-release to make modders aware of this change.

**Review by GraionDilach — DISMISSED** [review]
👍

**Comment by pchote** [comment]
> damaged frame of the civ buildings alters the bounds a bit. That's pretty much the only possible issue to look out for, though.

I have filed #15290 to address this, changing this PR to a "simple" polish improvement rather than a bugfix/bandaid.

File: mods/cnc/rules/ships.yaml [hunk 1]
```diff
@@ -70,6 +70,8 @@ LST:
 	RevealsShroud:
 		Range: 7c0
 	WithFacingSpriteBody:
+	Selectable:
+		Bounds: 48,40
```
Comment (pchote) [inline mods/cnc/rules/ships.yaml]: IMO this should be 48,48: this matches the original sprite, and looks better across the range of casings.

File: mods/cnc/rules/structures.yaml [hunk 2]
```diff
@@ -200,7 +206,7 @@ PROC:
 		PipCount: 10
 		Capacity: 700
 	Selectable:
-		Bounds: 72,56,0,12
+		Bounds: 72,56
```
Comment (pchote) [inline mods/cnc/rules/structures.yaml]: Why remove the offset here?

File: mods/ra/rules/fakes.yaml [hunk 3]
```diff
@@ -33,6 +35,8 @@ FPWR:
 SYRF:
 	Inherits: ^FakeBuilding
 	Inherits@infiltrate: ^InfiltratableFake
+	Selectable:
+		Bounds: 72,72
```
Comment (pchote) [inline mods/ra/rules/fakes.yaml]: This should be 72,48.

File: mods/ra/rules/structures.yaml [hunk 4]
```diff
@@ -242,6 +246,8 @@ SPEN:
 
 SYRD:
 	Inherits: ^Building
+	Selectable:
+		Bounds: 72,72
```
Comment (pchote) [inline mods/ra/rules/structures.yaml]: 72,48

**Review by pchote — COMMENTED** [review]
Just a couple of minor comments:

File: mods/cnc/rules/structures.yaml (hunk 2 above)
Comment (reaperrr) [reply mods/cnc/rules/structures.yaml]: See description of https://github.com/OpenRA/OpenRA/pull/15198/commits/77603d23d52b8de77e27f7c04e294d621ed5bfb4 for my reasoning. If you disagree I can drop this change, but to me the new value feels better.

**Comment by reaperrr** [comment]
Rebased and updated.

**Comment by pchote** [comment]
Checking against the last release, the (fake + normal) sub pen bounds should also be smaller.

**Comment by reaperrr** [comment]
> Checking against the last release, the (fake + normal) sub pen bounds should also be smaller.

Fixed.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/bc677865d108ec80bc1044d43d4d46c68c8b0781

**Pull request merged**